std/math/fma.zig

```diff
@@ -16,14 +23,15 @@ fn fma32(x: f32, y: f32, z: f32) f32 {
     const u = @bitCast(u64, xy_z);
     const e = (u >> 52) & 0x7FF;
 
-    if ((u & 0x1FFFFFFF) != 0x10000000 or e == 0x7FF or xy_z - xy == z) {
+    if ((u & 0x1FFFFFFF) != 0x10000000 or e == 0x7FF or (xy_z - xy == z and xy_z - z == xy)) {
```

Comment:
This seems pointless since the branches perform the same function currently, but it is just to match upstream. Specifically this patch: https://git.musl-libc.org/cgit/musl/commit/src/math?id=282b1cd26649d69de038111f5876853df6ddc345

As for why the second-branch isn't filled in, we don't handle different rounding modes as in-depth as musl and this is doesn't do anything different if using the default from memory.